File: backend/src/main/scala/microservices/operations/attraction/objects/UploadAttractionImagePlannerResponse.scala
package com.typesafe.travel.operations.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadAttractionImagePlannerResponse(
    assetId: String,
    publicUrl: String,
    originalFileName: String,
    mimeType: String,
    fileSize: Long
)
object UploadAttractionImagePlannerResponse:
  given sourceEncoder: Encoder[UploadAttractionImagePlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[UploadAttractionImagePlannerResponse] = deriveDecoder

File: frontend/src/microservices/operations/attraction/objects/UploadAttractionImagePlannerResponse.ts
// 本文件定义 attraction 模块的 `UploadAttractionImagePlannerResponse`，作为图片上传响应并提供 JSON 编解码。

export type UploadAttractionImagePlannerResponse = {
  assetId: string
  publicUrl: string
  originalFileName: string
  mimeType: string
  fileSize: number
}

export const uploadAttractionImagePlannerResponseFromJson = (json: string): UploadAttractionImagePlannerResponse =>
  JSON.parse(json) as UploadAttractionImagePlannerResponse

export const uploadAttractionImagePlannerResponseToJson = (value: UploadAttractionImagePlannerResponse): string =>
  JSON.stringify(value)
